Classes/Domain/Repository/PageRepository.php

```diff
@@ -3,8 +3,64 @@
 
 namespace WebVision\WvDeepltranslate\Domain\Repository;
 
+use TYPO3\CMS\Core\Database\Connection;
+use TYPO3\CMS\Core\Database\ConnectionPool;
 use TYPO3\CMS\Extbase\Persistence\Repository;
 
 final class PageRepository extends Repository
 {
+    private const TABLE_NAME = 'pages';
+
+    private ConnectionPool $connectionPool;
+
+    public function __construct()
+    {
+        $this->connectionPool = new ConnectionPool();
+    }
+
+    public function hasDeeplTranslatedContent()
+    {
+        $result = $this->connectionPool
+            ->getConnectionForTable(self::TABLE_NAME)
+            ->select(
+                ['uid', 'tx_wvdeepltranslate_has_translated_content'],
+                self::TABLE_NAME,
+                ['uid' => $this->getCurrentBackendPageId()]
+            )->fetchAssociative();
+
+        $deeplTrasalatedTime = $result['tx_wvdeepltranslate_has_translated_content'];
+        return $deeplTrasalatedTime > 0 ? true : false;
```

Comment:
Variable name is misspelled.

$result can also been given as 'false'. There is noch check, of a result is given in this context.

```suggestion
        if ($result !== false) {
            $deeplTranslatedTime = $result['tx_wvdeepltranslate_has_translated_content'];
            return ($deeplTranslatedTime > 0) ? true : false;
        }
```